lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts

```diff
@@ -84,6 +85,15 @@ export abstract class BaseManagedIdentitySource {
     ): ServerAuthorizationTokenResponse {
         let refreshIn, expiresIn: number | undefined;
         if (response.body.expires_on) {
+            // if the expires_on field in the response body is a string and in ISO 8601 format, convert it to a Unix timestamp (seconds since epoch)
+            if (
+                typeof response.body.expires_on === "string" &&
+                isIso8601(response.body.expires_on)
```

Comment:
Avoid double parsing. 

For example in C# the pattern used is `bool TryParseIso8601Date(string input, out Date parsedDate)`. I guess in NodeJS the pattern is more like `ParseIso8601(date: string) : Date | null` or smth like that (but not undefined)

Reply:
Not resolved. Double parsing happens at line 94. It's not a blocking comment.

Reply:
I still don't understand what you're asking. I reached out to you on Teams.